dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+from typing import TypeVar
+from dLux.utils import (get_positions_vector, get_pixel_positions)
+from abc import ABC, abstractmethod 
+import equinox as eqx
+import jax.numpy as np
+import jax 
+import functools
+
+jax.config.update("jax_enable_x64", True)
+
+Array = TypeVar("Array")
+Layer = TypeVar("Layer")
+Tensor = TypeVar("Tensor")
+Matrix = TypeVar("Matrix")
+Vector = TypeVar("Vector")
+
+
+__all__ = ["Aperture", "CompoundAperture", "SoftEdgedAperture", 
+    "SquareAperture", "SoftEdgedSquareAperture", "RectangularAperture",
+    "SoftEdgedRectangularAperture", "CircularAperture", 
+    "SoftEdgedCircularAperture", "AnnularAperture", 
+    "SoftEdgedAnnularAperture"]
+
+
+def cartesian_to_polar(coordinates : Tensor) -> Tensor:
+    """
+    Change the coordinate system from rectilinear to curvilinear.
+    
+    Parameters
+    ----------
+    coordinates : Tensor
+        The rectilinear coordinates.
+
+    Returns
+    -------
+    coordinates : Tensor
+        The curvilinear coordinates.
+    """
+    rho = np.hypot(coordinates[0], coordinates[1])
+    theta = np.arctan2(-coordinates[1], coordinates[0])
```

Comment:
Why the negative here?

Reply:
That was so that the coordinates fell in the way that I wanted them to. Otherwise they come out upside down. Hence why I implemented a second local function. This will probably be going any way I spoke to @benjaminpope and we want to switch from moving the coordinate system to moving the metric. I'll add a more detailed comment below.